one_fm/api/doc_methods/shift_request.py

```diff
@@ -331,16 +377,22 @@ def fill_to_date(doc, method):
         doc.to_date = doc.from_date
 
 def validate_from_date(doc, method):
-    if not doc.assign_day_off:
+    if doc.purpose != 'Assign Day Off' and not (frappe.session.user == get_employee_user_id(frappe.get_doc("ONEFM General Setting").get("attendance_manager"))):

```

Comment:
Instead of get_doc use get_single_value to fetch data from ONEFM General Settings